bochs/cpu/icache.cc

```diff
@@ -199,6 +220,8 @@ bxICacheEntry_c* BX_CPU_C::serveICacheMiss(Bit32u eipBiased, bx_phy_address pAdd
 
 bool BX_CPU_C::mergeTraces(bxICacheEntry_c *entry, bxInstruction_c *i, bx_phy_address pAddr)
 {
+  BX_ASSERT(!bx_dbg.debugger_active);
```

Comment:
mergeTraces should not be called with debugger active ever.
Also TODO: forming of traces is meaningless, may be we can just limit max length of a trace to 1 when debugger is running.